internal/services/storagecache/managed_lustre_file_system_auto_export_job_resource.go

```diff
@@ -0,0 +1,233 @@
+	AmlFileSystemName  string            `tfschema:"aml_file_system_name"`
+	Location           string            `tfschema:"location"`
+	AutoExportPrefixes []string          `tfschema:"auto_export_prefixes"`
+	AdminStatus        string            `tfschema:"admin_status"`
+	Tags               map[string]string `tfschema:"tags"`
+}
+
+type ManagedLustreFileSystemAutoExportJobResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagedLustreFileSystemAutoExportJobResource{}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ModelObject() interface{} {
+	return &ManagedLustreFileSystemAutoExportJobModel{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ResourceType() string {
+	return "azurerm_managed_lustre_file_system_auto_export_job"
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"aml_file_system_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedLustreFileSystemName,
+		},
+
+		"auto_export_prefixes": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"admin_status": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(autoexportjob.AutoExportJobAdminStatusEnable),
+				string(autoexportjob.AutoExportJobAdminStatusDisable),
+			}, false),
+			Default: string(autoexportjob.AutoExportJobAdminStatusEnable),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedLustreFileSystemAutoExportJobModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding model: %w", err)
+			}
+
+			autoExportJobsClient := metadata.Client.StorageCache.AutoExportJobs
+			autoExportJobClient := metadata.Client.StorageCache.AutoExportJob
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := autoexportjobs.NewAutoExportJobID(subscriptionId, model.ResourceGroupName, model.AmlFileSystemName, model.Name)
+
+			existing, err := autoExportJobsClient.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing Auto Export Job %s: %+v", id, err)
```

Comment:
Since Go v1.13 the standard for error wrapping is to use a `%w` instead of `%+v`

```suggestion
				return fmt.Errorf("checking for existing Auto Export Job %s: %w", id, err)
```